### ISSUE TITLE
CM instance with two nodes behind scenes Azure PaaS MicroCHAP failed to authenticate

### ISSUE THREAD

**Issue by aleksandartraja** [SitecoreUnicorn/Unicorn#265]
**Do you want to request a *feature* or report a *bug*?**
A workaround

**What is the current behavior?**
One CM instance, two nodes in behind (scalability) on Azure PaaS and tools are authenticated using a shared secret between the tool and the Sitecore server running Unicorn, which is relayed via CHAP+HMAC-SHA512.

**If the current behavior is a bug, please provide the steps to reproduce.**
When we received the token and place a sync operation the token or the signature is validated again and the validation fails because it was generated from the first node and validated in the second node. Placing the ARRAffinity cookies in the request does not solve the problem. 
In the code repository, we didn't see any usings of machine name or key in order to validate the signature.

**What is the expected behavior?**
Authentication to work on both nodes because they have the same secret key.

**Please mention your Sitecore version and Unicorn version.**
Sitecore 8.2.170407 and unicorn 4.0.0.0

**Comment by kamsar** [comment]
As far as I know multi node CM is no longer a supported configuration (by
Sitecore) due to cache syncing issues. MicroCHAP stores challenges issued
in the web database temporarily by default so it is cluster aware, and not
dependent on machine key in any way.

**Comment by aleksandartraja** [comment]
Thanks for the swift reply, for the multi node supported configuration ok, but for MicroCHAP when you say cluster aware, what do you mean by that?
I can see the challenges are stored in web db and they are the same for both nodes, but when the signature is created some of the factors are different for each node during the creation.
Which factor influence this?

**Comment by kamsar** [comment]
I mean it should work when pointed to a load balancer. Which factors are
you seeing different?

**Comment by aleksandartraja** [comment]
it should but it doesn't :). I have attached the unicorn test PowerShell script that we used to prove this. So in the Function Get-Challenge we used the ARRAffinity cookie from the one node and in the Function Invoke-StreamingWebRequest we use the ARRAffinity cookie from the other server and in this case, it fails with 401. If we use the one cookie for both of the functions it will run successfully.
